lib/datadog/tracing/contrib/pg/instrumentation.rb

```diff
@@ -0,0 +1,200 @@
+              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
+              Tracing.trace(Ext::SPAN_ASYNC_EXEC_PREPARED, service: service, resource: statement_name,
+                                                           type: Tracing::Metadata::Ext::SQL::TYPE) do |span|
+                annotate_span_with_query!(span, service)
+                # Set analytics sample rate
+                Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+
+                result = super(statement_name, params, *args)
+                annotate_span_with_result!(span, result)
+                result
+              end
+            end
+
+            def sync_exec(sql, *args)
+              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
+              Tracing.trace(Ext::SPAN_SYNC_EXEC, service: service, resource: sql,
+                                                 type: Tracing::Metadata::Ext::SQL::TYPE) do |span|
+                annotate_span_with_query!(span, service)
+                # Set analytics sample rate
+                Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+
+                result = super(sql, *args)
+                annotate_span_with_result!(span, result)
+                result
+              end
+            end
+
+            def sync_exec_params(sql, params, *args)
+              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
+              Tracing.trace(Ext::SPAN_SYNC_EXEC_PARAMS, service: service, resource: sql,
+                                                        type: Tracing::Metadata::Ext::SQL::TYPE) do |span|
+                annotate_span_with_query!(span, service)
+                # Set analytics sample rate
+                Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+
+                result = super(sql, params, *args)
+                annotate_span_with_result!(span, result)
+                result
+              end
+            end
+
+            def sync_exec_prepared(statement_name, params, *args)
+              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
+              Tracing.trace(Ext::SPAN_SYNC_EXEC_PREPARED, service: service, resource: statement_name,
+                                                          type: Tracing::Metadata::Ext::SQL::TYPE) do |span|
+                annotate_span_with_query!(span, service)
+                # Set analytics sample rate
+                Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+
+                result = super(statement_name, params, *args)
+                annotate_span_with_result!(span, result)
+                result
+              end
+            end
+
+            private
```

Comment:
A suggestion to remove code duplication. Let me know if this makes sense:

```suggestion
            def sync_exec_prepared(statement_name, params, *args)
              trace(Ext::SPAN_SYNC_EXEC_PREPARED, resource: statement_name) do
                super(statement_name, params, *args)
              end
            end

            private
            
            def trace(name, resource:)
              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]

              Tracing.trace(name, service: service, resource: resource, type: Tracing::Metadata::Ext::SQL::TYPE) do |span|
                annotate_span_with_query!(span, service)
                # Set analytics sample rate
                Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?

                result = yield
                annotate_span_with_result!(span, result)
                result
              end
            end
```

Reply:
thank you!! made the changes and tested and everything works like before